src/deroff.rs

```diff
@@ -455,115 +454,694 @@ impl Deroffer {
+        self.skip_char(1);
+
+        let s0 = &self.s[1..2];
+
+        match s0 {
+            "\\" => {
+                if self.str_at(1) == "\"" {
+                    self.condputs("\n");
+                    return true;
+                }
+            }
+            "[" => {
+                self.refer = true;
+                self.condputs("\n");
+                return true;
+            }
+            "]" => {
+                self.refer = false;
+                self.skip_char(1);
+                return self.text();
+            }
+            "." => {
+                self.r#macro = false;
+                self.condputs("\n");
+                return true;
+            }
+            _ => (),
+        };
+
+        self.nobody = false;
+        let s0s1 = self.s[..2].to_owned();
+
+        if self.g_macro_dispatch(s0s1.as_str()) {
+            return true;
+        }
+
+        if SKIP_HEADERS && self.nobody {
+            return true;
+        }
+
+        self.skip_leading_whitespace();
+
+        while !self.s.is_empty() && !self.is_white(0) {
+            self.skip_char(1);
+        }
+
+        self.skip_leading_whitespace();
+
+        loop {
+            if !self.quoted_arg() && !self.text_arg() {
+                if !self.s.is_empty() {
+                    // This is bad
+                    let s = &self.str_at(0).to_owned();
+                    self.condputs(s);
+                    self.skip_char(1);
+                } else {
+                    return true;
+                }
+            }
+        }
+    }
+
+    fn font(&mut self) -> bool {
+        if let Some(m) = self.g_re_font.find(&self.s.to_owned()) {
+            self.skip_char(m.end());
+            true
+        } else {
+            false
+        }
+    }
+
+    fn comment(&self) -> bool {
         unimplemented!()
     }
 
-    fn macro_other(&mut self) -> bool {
+    fn numreq(&mut self) -> bool {
+        // In the python, it has a check that is already handled in esc_char_backslash, which is
+        // the only place it gets called, so I'll omit that check here
+
+        // This is written as `self.macro += 1` in the source, but I dont know why
+        // it does the same thing (false -> true, true -> still true) :shrug:
+        // self.r#macro = true;
+        // Upon further investigation, this is the weirdest function ever
+        // This is just a state placeholder thing
+        let mut m = self.r#macro as u8 + 1;
+
+        self.skip_char(3);
+
+        // There's this, which has a comment explaining my thoughts right now
+        /*
+        while self.str_at(0) != "'" and self.esc_char():
+            pass  # Weird
+        */
+        // And I dont know what the purpose of it would even be, if you can tell, lmk
+
+        if self.str_at(0) == "\'" {
+            self.skip_char(1);
+        }
+
+        m -= 1;
+        self.r#macro = m != 0;
+
+        true
+    }
+
+    fn text_arg(&self) -> bool {
```

Comment:
Can you please help to add https://github.com/rust-dc/fish-manpage-completions/pull/73/files#diff-5b0804df3203e0b4fc4f245618d61d53L413-R437 as well if you are planning to rebase it with master, it was already fixed in master but I change it as well.